### ISSUE TITLE
Get reports return Forbidden for some reports

### ISSUE THREAD

**Issue by hieuntp2** [microsoftgraph/msgraph-sdk-dotnet#1751]
**Describe the bug**
Some graph API return forbidden error message below:
{"error":{"code":"MissingCredentials","message":"Valid Authorization header is needed."}}

**To Reproduce**
I'm using Microsoft graph to call API, also try to use Postman with an access token, but still have a same error message. The application have permission Report.Read.All too. I just have get this error mesage from Monday, but previous week is working fine.

**Comment by andrueastman** [comment]
Hey @hieuntp2 

Thanks for raising this. Any chance you are able to share the `request-id`/`client-request-id` from the error response so that we may share this with the relevant API team?

**Comment by hieuntp2** [comment]
Thank you for your response

Here is some information from Postman:
Request ID: 77A9ABAB9C0D493AB931802BB84D6BF0
Client Request ID: <sorry response doesn't contain>
URL: https://graph.microsoft.com/v1.0/reports/microsoft.graph.getOneDriveActivityUserCounts(period='D180')
![image](https://user-images.githubusercontent.com/4241455/226555293-048b305a-cdeb-49f1-9fc0-39020f3dc87f.png)


I tried to call the same API in Grap Explorer but return CORS, that weight. 
I

**Comment by Watto2022** [comment]
The same is happening when you invoke
`Invoke-WebRequest -Headers $authHeader1 -Uri "https://graph.microsoft.com/beta/reports/getOffice365ActiveUserDetail(period='D90')?$format=application/json"`
or run Graph powershell module command
`Get-MgReportOffice365ActiveUserDetail -Period 'D90' -OutFile ./Office365ActiveUserDetail.csv`

![image](https://user-images.githubusercontent.com/112093748/226558232-4abf55df-33e7-4aea-ad86-3e029bbd6222.png)

This for sure was working on 20 March 2023 and not working this morning 21 March 2023

**Comment by tuquy69** [comment]
I have seen this issue in some graphs API related to reporting count the same as you @hieuntp2 :triumph:, like below image. Not only Graph API OneDrive but also Yammer, Exchange, Skype Business, and Teams,... are happening this week although the previous week, they are working fine! :disappointed_relieved:

![Capture](https://user-images.githubusercontent.com/114461984/226570420-b8a8c374-add6-45a7-a1ff-ab320d1152e3.PNG)

**Comment by andrueastman** [comment]
@hieuntp2 @tuquy69 As you are using postman, any chance you can generate a GUID and send it in the header `client-request-id` so as the API team may be able to use it to co-relate the error?
https://learn.microsoft.com/en-us/graph/best-practices-concept#reliability-and-support

**Comment by andrueastman** [comment]
We've forwarded this to the relevant API team and will give feedback once we get information.

https://portal.microsofticm.com/imp/v3/incidents/details/376433806/home

**Comment by Watto2022** [comment]
It looks like the problem is resolved this morning.
![image](https://user-images.githubusercontent.com/112093748/226850277-d5e755a5-39a7-46b5-b2d4-37a501d84576.png)
We are able to authenticate to the https://graph.microsoft.com/beta/reports/getOffice365ActiveUserDetail using certificates.
Just to mentioned we ran test yesterday with Microsoft support engineer and we were able to successfully authenticate using client secret instead of using client certificate. However this was not a solution as we have too many scripts to be changed from authenticating using certificate to client secret.
